stubs/ldap3/ldap3/protocol/rfc4511.pyi

```diff
@@ -1,32 +1,26 @@
-from typing import Any as _Any
-
-# Enable when pyasn1 gets stubs:
-# from pyasn1.type.univ import Boolean, Choice, Enumerated, Integer, Null, OctetString, Sequence, SequenceOf, SetOf
-Boolean = _Any
-Choice = _Any
-Enumerated = _Any
-Integer = _Any
-Null = _Any
-OctetString = _Any
-Sequence = _Any
-SequenceOf = _Any
-SetOf = _Any
-
-LDAP_MAX_INT: int
-MAXINT: _Any
-rangeInt0ToMaxConstraint: _Any
-rangeInt1To127Constraint: _Any
-size1ToMaxConstraint: _Any
-responseValueConstraint: _Any
-numericOIDConstraint: _Any
-distinguishedNameConstraint: _Any
-nameComponentConstraint: _Any
-attributeDescriptionConstraint: _Any
-uriConstraint: _Any
-attributeSelectorConstraint: _Any
+from typing_extensions import Final as _Final
+
+from pyasn1.type.constraint import ConstraintsIntersection, SingleValueConstraint, ValueRangeConstraint, ValueSizeConstraint
+from pyasn1.type.namedtype import NamedTypes
+from pyasn1.type.namedval import NamedValues
+from pyasn1.type.tag import TagSet
+from pyasn1.type.univ import Boolean, Choice, Enumerated, Integer, Null, OctetString, Sequence, SequenceOf, SetOf
+
+LDAP_MAX_INT: _Final = 2147483647  # noqa: Y015  # name conflict
+MAXINT: _Final[Integer]
+rangeInt0ToMaxConstraint: ValueRangeConstraint
+rangeInt1To127Constraint: ValueRangeConstraint
+size1ToMaxConstraint: ValueSizeConstraint
+responseValueConstraint: SingleValueConstraint
+numericOIDConstraint: None
+distinguishedNameConstraint: None
+nameComponentConstraint: None
+attributeDescriptionConstraint: None
+uriConstraint: None
+attributeSelectorConstraint: None
```

Comment:
Hmm, I struggle to believe that these are _always_ meant to be None, otherwise that's just a bit pointless having them. Maybe this?

```suggestion
numericOIDConstraint: Incomplete | None
distinguishedNameConstraint: Incomplete | None
nameComponentConstraint: Incomplete | None
attributeDescriptionConstraint: Incomplete | None
uriConstraint: Incomplete | None
attributeSelectorConstraint: Incomplete | None
```

Reply:
Source:
https://github.com/cannatag/ldap3/blob/dev/ldap3/protocol/rfc4511.py#L60
```py
# custom constraints
numericOIDConstraint = None  # TODO
distinguishedNameConstraint = None  # TODO
nameComponentConstraint = None  # TODO
attributeDescriptionConstraint = None  # TODO
uriConstraint = None  # TODO
attributeSelectorConstraint = None  # TODO
```

I'm pretty sure they're None because they have yet to add the correct constraint. They are only "used" below in the same module in commented code.

Typing it as None seems better than omitting and adding an allowlist entry: if/when they are implemented, stubsabot can pick up the change.

Reply:
Hmm, okay. It's pretty unusual, though -- can we add a comment to clarify why we're typing it this way? :)